File: androidDemo/src/main/java/com/getyourmap/androidDemo/NumberFormatter.java
package com.getyourmap.androidDemo;

import java.util.Locale;

public class NumberFormatter {
  static Locale locale = Locale.getDefault();

  public static String NiceDoubleToString(double val) {
    if (Double.isNaN(val)) {
      return "---";
    }

    if (val < 10) {
      return String.format(locale, "%.2f", val);
    } else if (val < 100) {
      return String.format(locale, "%.1f", val);
    }
    return String.format(locale, "%.0f", val);
  }

  public static String FormatSize(long val) {
    double sizeInMB = (double) val / (1000 * 1000);
    return String.format("%s %s", NiceDoubleToString(sizeInMB), "MB");
  }
}

File: kotlinDemo/src/main/java/com/getyourmap/androidDemo/NumberFormatter.kt
package com.getyourmap.androidDemo

import java.util.*

internal object NumberFormatter {
    private var locale: Locale = Locale.getDefault()
    private fun Double.toNiceString(): String {
        if (this.isNaN()) {
            return "---"
        }
        if (this < 10) {
            return String.format(locale, "%.2f", this)
        } else if (this < 100) {
            return String.format(locale, "%.1f", this)
        }
        return String.format(locale, "%.0f", this)
    }

    fun formatSize(value: Long): String {
        val sizeInMB = value.toDouble() / (1000 * 1000)
        return "${sizeInMB.toNiceString()} MB"
    }
}